File: swift/prime-factors/Sources/PrimeFactors/PrimeFactors.swift
struct PrimeFactors {
    let input: Int
    
    var toArray: [Int] {
        var factors = [Int]()
        var n = input
        var testVal = 2
        while n > 1 {
            while (n % testVal == 0) {
                factors.append(testVal)
                n /= testVal
            }
            testVal += 1
        }
        return factors
    }
    
    init(_ input: Int) {
        self.input = input
    }
}

File: kotlin/prime-factors/src/main/kotlin/PrimeFactors.kt
object PrimeFactorCalculator {

    fun primeFactors(int: Int): List<Int> = primeFactors(int.toLong()).map { it.toInt() }

    fun primeFactors(long: Long): List<Long> {
        var factors = mutableListOf<Long>()
        var testVal = 2L
        var n = long
        while (n > 1) {
            while (n % testVal == 0L) {
                factors.add(testVal)
                n /= testVal
            }
            testVal++
        }
        return factors
    }
}
